Arrays do not match in python/vm_types_test:test_variant_list_buffers on Windows

Spotted when adding a Windows CI build: https://github.com/iree-org/iree/actions/runs/3414442364/jobs/5682427788, and then reproduced locally.

Full logs:
```
    Start 1225: iree/runtime/bindings/python/vm_types_test
1225: Test command: C:\Python38\python.exe "D:/dev/projects/iree/runtime/bindings/python/tests/vm_types_test.py"
1225: Working Directory: D:/dev/projects/iree-build/runtime/bindings/python
1225: Environment variables: 
1225:  PYTHONPATH=D:/dev/projects/iree-build/compiler/bindings/python;D:/dev/projects/iree-build/runtime/bindings/python;
1225:  TEST_TMPDIR=D:/dev/projects/iree-build/test_tmpdir/iree/runtime/bindings/python/vm_types_test_test_tmpdir
1225: Test timeout computed to be: 10000000
1225: .INFO:root:variant_list: <VmVariantList(0): []>
1225: .F....
1225: ======================================================================
1225: FAIL: test_variant_list_buffers (__main__.VmTypesTest)
1225: ----------------------------------------------------------------------
1225: Traceback (most recent call last):
1225:   File "D:/dev/projects/iree/runtime/bindings/python/tests/vm_types_test.py", line 70, in test_variant_list_buffers
1225:     np.testing.assert_array_equal(ary1, ary2)
1225:   File "C:\Python38\lib\site-packages\numpy\testing\_private\utils.py", line 934, in assert_array_equal
1225:     assert_array_compare(operator.__eq__, x, y, err_msg=err_msg,
1225:   File "C:\Python38\lib\site-packages\numpy\testing\_private\utils.py", line 844, in assert_array_compare
1225:     raise AssertionError(msg)
1225: AssertionError: 
1225: Arrays are not equal
1225: 
1225: Mismatched elements: 3 / 4 (75%)
1225: Max absolute difference: 4
1225: Max relative difference: 0.5
1225:  x: array([1, 2, 3, 4], dtype=int64)
1225:  y: array([1, 0, 2, 0], dtype=int64)
1225: 
1225: ----------------------------------------------------------------------
1225: Ran 7 tests in 0.005s
1225: 
1225: FAILED (failures=1)
1225: <VmRef vm.list at 0x1a8a7d04ba0>
1225: <VmRef vm.list at 0x1a8a7d04ba0>
1225: <VmVariantList(0): []>
1225: <VmVariantList(0): []>
1225: <VmBuffer size 10 at 0x1a8a7d05030>
4/9 Test #1225: iree/runtime/bindings/python/vm_types_test ........***Failed    0.66 sec
```

Test code: https://github.com/iree-org/iree/blob/75636a94541ccd42e170c0bb72bf50a31b2a6cae/runtime/bindings/python/tests/vm_types_test.py#L49-L72

--- a/build_tools/cmake/ctest_all.sh
+++ b/build_tools/cmake/ctest_all.sh
@@ -90,8 +90,6 @@ if [[ "$OSTYPE" =~ ^msys ]]; then
     "iree/tests/e2e/tensor_ops/check_vmvx_ukernel_local-task_unpack.mlir"
     # TODO(#11070): Fix argument/result signature mismatch
     "iree/tests/e2e/tosa_ops/check_vmvx_local-sync_microkernels_fully_connected.mlir"
-    # TODO(#11080): Fix arrays not matching in test_variant_list_buffers
-    "iree/runtime/bindings/python/vm_types_test"
   )
 elif [[ "$OSTYPE" =~ ^darwin ]]; then
   excluded_tests+=(

--- a/runtime/bindings/python/hal.cc
+++ b/runtime/bindings/python/hal.cc
@@ -411,54 +411,59 @@ HalDevice HalDriver::CreateDeviceByURI(std::string& device_uri,
 namespace {
 
 py::object MapElementTypeToDType(iree_hal_element_type_t element_type) {
-  // See: https://docs.python.org/3/c-api/arg.html#numbers
-  // TODO: Handle dtypes that do not map to a code (i.e. fp16).
-  const char* dtype_code;
+  // See:
+  //   * https://numpy.org/doc/stable/reference/arrays.dtypes.html
+  //   * https://docs.python.org/3/c-api/arg.html#numbers
+  //
+  // Single letter codes can be ambiguous across platforms, so prefer explicit
+  // bit depth values, ("Type strings: Any string in numpy.sctypeDict.keys()").
+  // See https://github.com/pybind/pybind11/issues/1908
+  const char* dtype_string;
   switch (element_type) {
     case IREE_HAL_ELEMENT_TYPE_BOOL_8:
-      dtype_code = "?";
+      dtype_string = "?";
       break;
     case IREE_HAL_ELEMENT_TYPE_INT_8:
     case IREE_HAL_ELEMENT_TYPE_SINT_8:
-      dtype_code = "b";
+      dtype_string = "int8";
       break;
     case IREE_HAL_ELEMENT_TYPE_UINT_8:
-      dtype_code = "B";
+      dtype_string = "uint8";
       break;
     case IREE_HAL_ELEMENT_TYPE_INT_16:
     case IREE_HAL_ELEMENT_TYPE_SINT_16:
-      dtype_code = "h";
+      dtype_string = "int16";
       break;
     case IREE_HAL_ELEMENT_TYPE_UINT_16:
-      dtype_code = "H";
+      dtype_string = "uint16";
       break;
     case IREE_HAL_ELEMENT_TYPE_INT_32:
     case IREE_HAL_ELEMENT_TYPE_SINT_32:
-      dtype_code = "i";
+      dtype_string = "int32";
       break;
     case IREE_HAL_ELEMENT_TYPE_UINT_32:
-      dtype_code = "I";
+      dtype_string = "uint32";
       break;
     case IREE_HAL_ELEMENT_TYPE_INT_64:
     case IREE_HAL_ELEMENT_TYPE_SINT_64:
-      dtype_code = "l";
+      dtype_string = "int64";
       break;
     case IREE_HAL_ELEMENT_TYPE_UINT_64:
-      dtype_code = "L";
+      dtype_string = "uint64";
       break;
     case IREE_HAL_ELEMENT_TYPE_FLOAT_16:
-      dtype_code = "e";
+      dtype_string = "float16";
       break;
     case IREE_HAL_ELEMENT_TYPE_FLOAT_32:
-      dtype_code = "f";
+      dtype_string = "float32";
       break;
     case IREE_HAL_ELEMENT_TYPE_FLOAT_64:
-      dtype_code = "d";
+      dtype_string = "float64";
       break;
     default:
       throw RaiseValueError("Unsupported VM Buffer -> numpy dtype mapping");
   }
-  return py::dtype(dtype_code);
+  return py::dtype(dtype_string);
 }
 
 }  // namespace

--- a/runtime/bindings/python/tests/vm_types_test.py
+++ b/runtime/bindings/python/tests/vm_types_test.py
@@ -49,12 +49,17 @@ class VmTypesTest(unittest.TestCase):
   def test_variant_list_buffers(self):
     device = rt.get_device("local-sync")
     ET = rt.HalElementType
-    for dt, et in ((np.int8, ET.SINT_8), (np.int16, ET.SINT_16),
-                   (np.int32, ET.SINT_32), (np.int64, ET.SINT_64),
-                   (np.uint8, ET.UINT_8), (np.uint16, ET.UINT_16),
-                   (np.uint32, ET.UINT_32), (np.uint64, ET.UINT_64),
-                   (np.float32, ET.FLOAT_32), (np.float64, ET.FLOAT_64)):
-      # TODO: Unimplemented: (np.float16, ET.FLOAT_16)
+    for dt, et in ((np.int8, ET.SINT_8),      #
+                   (np.int16, ET.SINT_16),    #
+                   (np.int32, ET.SINT_32),    #
+                   (np.int64, ET.SINT_64),    #
+                   (np.uint8, ET.UINT_8),     #
+                   (np.uint16, ET.UINT_16),   #
+                   (np.uint32, ET.UINT_32),   #
+                   (np.uint64, ET.UINT_64),   #
+                   (np.float16, ET.FLOAT_16), #
+                   (np.float32, ET.FLOAT_32), #
+                   (np.float64, ET.FLOAT_64)):
       lst = rt.VmVariantList(5)
       ary1 = np.asarray([1, 2, 3, 4], dtype=dt)
       bv1 = device.allocator.allocate_buffer_copy(
@@ -65,7 +70,6 @@ class VmTypesTest(unittest.TestCase):
       lst.push_ref(bv1)
       ary2 = rt.DeviceArray(device,
                             lst.get_as_object(0, rt.HalBufferView),
-                            override_dtype=dt,
                             implicit_host_transfer=True)
       np.testing.assert_array_equal(ary1, ary2)
       with self.assertRaises(IndexError):

--- a/runtime/bindings/python/tests/vm_types_test.py
+++ b/runtime/bindings/python/tests/vm_types_test.py
@@ -49,17 +49,18 @@ class VmTypesTest(unittest.TestCase):
   def test_variant_list_buffers(self):
     device = rt.get_device("local-sync")
     ET = rt.HalElementType
-    for dt, et in ((np.int8, ET.SINT_8),      #
-                   (np.int16, ET.SINT_16),    #
-                   (np.int32, ET.SINT_32),    #
-                   (np.int64, ET.SINT_64),    #
-                   (np.uint8, ET.UINT_8),     #
-                   (np.uint16, ET.UINT_16),   #
-                   (np.uint32, ET.UINT_32),   #
-                   (np.uint64, ET.UINT_64),   #
-                   (np.float16, ET.FLOAT_16), #
-                   (np.float32, ET.FLOAT_32), #
-                   (np.float64, ET.FLOAT_64)):
+    for dt, et in (
+        (np.int8, ET.SINT_8),  #
+        (np.int16, ET.SINT_16),  #
+        (np.int32, ET.SINT_32),  #
+        (np.int64, ET.SINT_64),  #
+        (np.uint8, ET.UINT_8),  #
+        (np.uint16, ET.UINT_16),  #
+        (np.uint32, ET.UINT_32),  #
+        (np.uint64, ET.UINT_64),  #
+        (np.float16, ET.FLOAT_16),  #
+        (np.float32, ET.FLOAT_32),  #
+        (np.float64, ET.FLOAT_64)):
       lst = rt.VmVariantList(5)
       ary1 = np.asarray([1, 2, 3, 4], dtype=dt)
       bv1 = device.allocator.allocate_buffer_copy(

